### PR TITLE
Dex trades Fraxswap Avalanche (C-Chain)

### PR DESCRIPTION
**Pull request by zhongyiio**
Brief comments on the purpose of your changes:

Fraxswap trades on Avalanche (C-Chain). Also, add FRAX and FXS tokens to prices.usd.

*For Dune Engine V2*
I've checked that:
General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
* [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Review by 0xBoxer — CHANGES_REQUESTED** [review]
left some comments in line, please work with the decoded contract data once it is decoded.

also for some reason our 'WAVAX' price insert is broken, I'll bring this up with the team internally.

File: models/fraxswap/avalanche_c/fraxswap_avalanche_c_trades.sql [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        ,t.call_trace_address
+        ,t.call_tx_hash
+        ,t.contract_address
+        ,t.`to`
+    FROM fraxswap_decodes t
+    INNER JOIN {{ source('avalanche_c', 'logs') }} l
+        ON t.call_tx_hash = l.tx_hash
+        AND l.topic1 = "0xd78ad95fa46c994b6551d0da85fc275fe613ce37657fb8d5e3d130840159d822"
+        {% if is_incremental() %}
+        AND l.block_time >= date_trunc("day", now() - interval '1 week')
```
[the remark this answers is not in the document]
Comment (zhongyiio) [reply models/fraxswap/avalanche_c/fraxswap_avalanche_c_trades.sql:164]: I found some related contracts in the documentation and submitted them. I have seen this in Ethereum, but I didn't find the contract address and thought it didn't exist.

Haha, thank you for your help to submit, and I will modify it when the contract is decoded.

image

**Comment by augustog** [comment]
FYI @0xBoxer the failing test is a known issue on our side, it doesn't detect jinja variables as valid refs, we can safely ignore that error, but we should validate metadata a bit more closely.

**Comment by 0xBoxer** [comment]
hey @jeff-dude this is ready to be merged

**Review by 0xBoxer — APPROVED** [review]
great stuff, thanks for implementing the changes @zhongyiio

**Review by jeff-dude — APPROVED** [review]

**Comment by jeff-dude** [comment]
for reference, until we enhance our generic test for token bought, i like to check the query without false filter to ensure join is working as expected:
```
with
  unit_test as (
    select
      case
        when lower(m.token_bought_address) = lower(seed.token_bought_address) then True
        else False
      end as token_bought_test
    from
      test_schema.git_edae8c1d_fraxswap_avalanche_c_trades m
      inner join test_schema.git_edae8c1d_dex_trades_seed seed on m.tx_hash = seed.tx_hash
      and m.evt_index = seed.evt_index
      and m.block_date = seed.block_date
      and m.blockchain = seed.blockchain
      and m.project = seed.project
      and m.version = seed.version
  )
select
  *
from
  unit_test
```

future state of the generic test will ideally ensure the output row count from join matches the number of entries in seed file for that dex. this shows 8 results, all true, matching the count in seed file.

**Pull request merged**